src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -1,15 +1,18 @@
 from PyQt5.QtCore import QRect, Qt
 from PyQt5.QtWidgets import QWidget, QGroupBox, QLabel, QComboBox, QVBoxLayout, QPushButton, QCheckBox, \
-    QMessageBox, QSplashScreen, QApplication, QDesktopWidget, QFormLayout, QHBoxLayout, QSizePolicy, QScrollArea
+    QMessageBox, QSplashScreen, QApplication, QDesktopWidget, QFormLayout, QHBoxLayout, QSizePolicy, QScrollArea, \
+    QSpinBox
 
 from widgets import UnfoldWidget
+from widgets.tables import DataPreviewScreen, PreviewReason
 
 
 class PreprocessingWidget(UnfoldWidget):
     def __init__(self, parent: QWidget, engine):
         super().__init__(parent, engine, 'preprocessing_widget', 'PREPROCESSING')
 
         self.data_submitted = False
+        self.reduced_columns = []
```

Comment:
I think we should have this information in the State and clear it, when we load new data, because we drop this columns from teh raw_data so it may cause removing some data from the new dataset.
When I load new data and click Reduce button, the app crashes.